activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -54,16 +54,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
```

Comment:
That is as change in the behavior because before you could mutate the values. Also I don't see any reason to not mutate the values, so I prefer to not do that.

BTW, this PR already turned out in a huge bikesheding, your opinions against the core member opinions. Because of that I'm wondering if the performance improvement of this PR (if any) is worth the time we are discussion this.